ot/mpz-ot/src/kos/shared_receiver.rs

```diff
@@ -0,0 +1,55 @@
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use itybity::IntoBitIterator;
+use mpz_common::{sync::Mutex, Context};
+use mpz_core::Block;
+use serio::{stream::IoStreamExt, SinkExt};
+use utils_aio::non_blocking_backend::{Backend, NonBlockingBackend};
+
+use crate::{
+    kos::{Receiver, ReceiverError},
+    OTError, OTReceiver,
+};
+
+/// A shared KOS receiver.
+#[derive(Debug, Clone)]
+pub struct SharedReceiver<BaseOT> {
```

Comment:
Shouldn't these follow the new `Syncer` and `Mutex` approach?

Reply:
I realized that now that they do :sweat_smile: 

Can we eventually rename the 'Mutex'. I am open to suggestions. What about `MpcMutex`?